Translates `responding-to-events` page (#684)
reactjs/pt-br.react.dev: src/content/learn/responding-to-events.md

---
title: Respondendo a Eventos
---

<Intro>

> O React permite você adicionar *manipuladores de eventos* (event handler) ao seu JSX. Os manipuladores de eventos são funções independentes que são acionadas em resposta a interações como clicar com o mouse, passar o cursor do mouse sobre um certo elemento, focar em campos de formulário, entre outros.

Reviewer: ```suggestion
O React permite que você adicione *manipuladores de evento* a sua JSX. Os manipuladores de eventos funções independentes as quais são acionadas em resposta a interações como cliques, passagem do cursor sobre certo elemento, mudança do foco em campos de formulário, entre outros.
```

</Sandpack>

> Isso permite que esses dois botões exibem mensagens diferentes. Tente alterar as mensagens passadas para eles.

Reviewer 2: ```suggestion
Isso permite que esses dois botões exibam mensagens diferentes. Tente alterar as mensagens passadas para eles.
```

### Passando manipuladores de eventos como props {/*passing-event-handlers-as-props*/}

> É comum que o componente pai especifique o manipulador de eventos de um componente filho. Por exemplo, considere os botões: dependendo do contexto em que o componente `Button` é usado, pode ser necessário executar funções diferentes, talvez um reproduza um filme e outro faça o upload de uma imagem.

Reviewer 2: ```suggestion
É comum que o componente pai defina o manipulador de eventos de um componente filho. Por exemplo, considere os botões: dependendo do contexto em que o componente `Button` é usado, pode ser necessário executar funções diferentes, talvez um reproduza um filme e outro faça o upload de uma imagem.
```

</Sandpack>

Aqui, o componente `Toolbar` renderiza o componente `PlayButton` e o componente`UploadButton`:

- O `PlayButton` passa o `handlePlayClick` como prop `onClick` para o `Button` dentro dele.
- O `UploadButton` passa `() => alert('Enviando!')` como a prop `onClick` para o `Button` dentro.

> Por fim, seu componente `Button` aceita uma prop chamada `onClick`. Ele passa essa prop diretamente para a tag `<button>` nativa do navegador usando `onClick={onClick}`. Isso diz ao React para chamar a função quando o botão for clicado.

Reviewer 2: ```suggestion
Por fim, seu componente `Button` aceita uma prop chamada `onClick`. Ele passa essa prop diretamente para o elemento `<button>` nativo do navegador usando `onClick={onClick}`. Isso diz ao React para chamar a função quando o botão for clicado.
```

Se você usa um [design system](https://uxdesign.cc/everything-you-need-to-know-about-design-systems-54b109851969), é comum que componentes, como botões, contenham estilo mas não especifiquem o comportamento. Em vez disso, componentes como `PlayButton` e `UploadButton` passarão os manipuladores de eventos para baixo.

### Nomeando props de manipuladores de eventos {/*naming-event-handler-props*/}

Os componentes nativos, como `<button>` e `<div>`, suportam apenas [os nomes de eventos do navegador](/reference/react-dom/components/common#common-props), tais como `onClick`. No entanto, quando você está criando seus próprios componentes, você pode nomear os manipuladores de eventos da forma que preferir.

> Por convenção, as props dos manipuladores de eventos devem começar com `on`, seguido por uma letra maiúscula.

Reviewer 2: ```suggestion
Por convenção, as props dos manipuladores de eventos devem começar com o termo `on`, seguido por uma letra maiúscula.
```
